src/Microsoft.AspNetCore.Razor.Language/Legacy/SpanEditHandler.cs

```diff
@@ -82,42 +82,39 @@ protected virtual SpanBuilder UpdateSpan(Span target, TextChange normalizedChang
             return newSpan;
         }
 
-        protected internal static bool IsAtEndOfFirstLine(Span target, TextChange change)
+        public string ApplyChange(SourceChange change, string content, int changeOffset)
+        {
+            var changeRelativePosition = change.Span.AbsoluteIndex - changeOffset;
+
+            Debug.Assert(changeRelativePosition >= 0);
+            return content.Remove(changeRelativePosition, change.Span.Length).Insert(changeRelativePosition, change.NewText);
```

Comment:
Could we make this bit more debuggable. Have a feeling we might be debugging through here in the future:
```C#
var appliedChange = content.Remove(changeRelativePosition, change.Span.Length).Insert(changeRelativePosition, change.NewText);
return appliedChange;
```